clang/include/clang/Tooling/DependencyScanning/DependencyScanningWorker.h

```diff
@@ -111,6 +112,7 @@ class DependencyScanningWorker {
   /// Whether to optimize the modules' command-line arguments.
   bool OptimizeArgs;
 
+  const CASOptions &CASOpts;
```

Comment:
I would suggest copying the value or making it a `shared_ptr` if you think it's too expensive to copy.  The libclang documentation says the service should outlive the workers, but it's not obvious to me that it shouldn't be allowed to go away earlier, and if someone gets it wrong it's going to be UAF potentially deep inside the worker, and might be hard to diagnose.

Reply:
I've changed it to a copy.👍